### ISSUE TITLE
[ Bynary Protocol ] When a wrong authentication token is provided the response lacks a field and break the protocol

### ISSUE THREAD

**Issue by Ostico** [orientechnologies/orientdb#4764]
A simple example when a token created by a user ( admin , admin ) is used by another user ( root, root ):
```php
require('vendor/autoload.php');
$client = new \PhpOrient\PhpOrient('localhost', 2424);
$client->setSessionToken( true );
$client->dbOpen( "GratefulDeadConcerts", 'admin', 'admin' );
$admin_db_token = $client->getSessionToken();
unset($client);

$client = new \PhpOrient\PhpOrient('localhost', 2424);
$client->username = 'root';
$client->password = 'root';
$client->connect();
$client->setSessionToken( $admin_db_token );
# Expected exception
$client->dbExists("GratefulDeadConcerts");
```

```
# The token is valid but it is not of this user ( wrong user ), please note the [] field

INFO - Writing byte (1 byte): 1 [OChannelBinaryServer]
INFO - Writing int (4 bytes): -1 [OChannelBinaryServer]
INFO - Writing bytes (4+0=4 bytes): [] [OChannelBinaryServer]
INFO - Writing byte (1 byte): 1 [OChannelBinaryServer]
INFO - Writing string (4+68=72 bytes): com.orientechnologies.orient.core.exception.OSecurityAccessException [OChannelBinaryServer]
INFO - Writing string (4+30=34 bytes): Server user not authenticated. [OChannelBinaryServer]
INFO - Writing byte (1 byte): 0 [OChannelBinaryServer]
INFO - Writing bytes (4+1383=1387 bytes): ........................
INFO - Flush [OChannelBinaryServer]
FINE - Sent run-time exception to the client /127.0.0.1:43706: com.orientechnologies.orient.core.exception.OSecurityAccessException: Server user not authenticated. [ONetworkProtocolBinary]
```

In this case i provide a WRONG token :

```php
// start a new connection
$client = new PhpOrient( 'localhost', 2424 );
$client->setSessionToken(  "WRONG_TOKEN" );

//if the token were correct, the dbOpen is not needed to perform database operations
$client->query( 'select * from V limit 10' );
```

```
# The token is wrong because it does not exists, please note the lacks of [] field

INFO - Writing byte (1 byte): 1 [OChannelBinaryServer]
INFO - Writing int (4 bytes): -1 [OChannelBinaryServer]
INFO - Writing byte (1 byte): 1 [OChannelBinaryServer]
INFO - Writing string (4+62=66 bytes): com.orientechnologies.orient.core.exception.OSecurityException [OChannelBinaryServer]
INFO - Writing string (4+64=68 bytes): The token provided is not a valid token, signature doesn't match [OChannelBinaryServer]
INFO - Writing byte (1 byte): 0 [OChannelBinaryServer]
INFO - Flush [OChannelBinaryServer]
FINE - Sent run-time exception to the client /127.0.0.1:43851: com.orientechnologies.orient.core.exception.OSecurityException: The token provided is not a valid token, signature doesn't match [ONetworkProtocolBinary]
INFO - Flush [OChannelBinaryServer]
```

The exception is raised but a field lacks in the response causing the client to hang because it wait for the token field to read.

**Comment by tglman** [comment]
hi @Ostico,

this should be fixed in 2.2

**Comment by Ostico** [comment]
Tnx @tglman ,
I will add this to the tests as soon as possible

**Comment by tglman** [comment]
hi,
Any news about this ? can i close ?

**Comment by tglman** [comment]
The PHP client was released with the support to 2.2 so i can consider this closed.
